File: examples/java/BulkTask.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

// 查询一个已提交的 ws_business_batch 任务。提交用 multipart，见 README 的 curl 示例。
public class BulkTask {
    public static void main(String[] args) throws Exception {
        String apiKey = System.getenv("WALOOKUP_API_KEY");
        if (apiKey == null) throw new IllegalStateException("Set WALOOKUP_API_KEY");
        String taskId = args.length > 0 ? args[0] : System.getenv("TASK_ID");
        if (taskId == null) throw new IllegalStateException("Pass a task id");
        HttpRequest request = HttpRequest.newBuilder(URI.create("https://walookup.com/api/v1/bulk-tasks/" + taskId))
            .header("X-API-Key", apiKey)
            .GET()
            .build();
        HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
        if (response.statusCode() >= 300) throw new IllegalStateException(response.body());
        System.out.println(response.body());
    }
}

File: examples/csharp/BulkTask.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

// 查询一个已提交的 ws_business_batch 任务。提交用 multipart，见 README 的 curl 示例。
class BulkTask
{
    static async Task Main(string[] args)
    {
        var apiKey = Environment.GetEnvironmentVariable("WALOOKUP_API_KEY")
            ?? throw new Exception("Set WALOOKUP_API_KEY");
        var taskId = args.Length > 0 ? args[0] : Environment.GetEnvironmentVariable("TASK_ID")
            ?? throw new Exception("Pass a task id");
        using var client = new HttpClient();
        client.DefaultRequestHeaders.Add("X-API-Key", apiKey);
        var response = await client.GetAsync($"https://walookup.com/api/v1/bulk-tasks/{taskId}");
        var body = await response.Content.ReadAsStringAsync();
        response.EnsureSuccessStatusCode();
        Console.WriteLine(body);
    }
}
